react/SwitcherListV2.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useRef } from 'react'
+import classnames from 'classnames'
+import { useCssHandles } from 'vtex.css-handles'
+
+import useEnhancedEffect from './modules/useEnhancedEffect'
+import ownerDocument from './modules/ownerDocument'
+
+interface Props {
+  children: React.ReactNode
+}
+
+const CSS_HANDLES = ['list'] as const
+
+type ChangeFocusFn = (
+  list: HTMLUListElement,
+  currentFocus: Element | null | undefined
+) => Element | null
+
+const nextItem: ChangeFocusFn = (list, currentFocus) => {
+  if (list === currentFocus) {
+    return list.firstChild as Element
+  }
+  if (currentFocus?.nextElementSibling) {
+    return currentFocus.nextElementSibling
+  }
+  return list.firstChild as Element
+}
+
+const previousItem: ChangeFocusFn = (list, currentFocus) => {
+  if (list === currentFocus) {
+    return list.lastChild as Element
+  }
+  if (currentFocus?.previousElementSibling) {
+    return currentFocus.previousElementSibling
+  }
+  return list.lastChild as Element
+}
+
+function moveFocus(
+  list: HTMLUListElement,
+  currentFocus: HTMLElement | null | undefined,
+  changeFocusFn: ChangeFocusFn
+) {
+  let passedThroughFirst = false
+  let nextFocus = changeFocusFn(list, currentFocus)
+  while (nextFocus) {
+    // Prevent infinite loop
+    if (nextFocus === list.firstChild) {
+      if (passedThroughFirst) {
+        return false
+      }
+      passedThroughFirst = true
+    }
+
+    if (!nextFocus.hasAttribute('tabindex') || (nextFocus as any).disabled) {
+      nextFocus = changeFocusFn(list, nextFocus)
+    } else {
+      ;(nextFocus as HTMLElement).focus()
+      return true
+    }
+  }
+}
+
+export default function SwitcherListV2(props: Props) {
```

Comment:
Why the `V2` here and in the file name? There is no V1, so I don't think it's necessary.

Reply:
I couldn't find a good name for it, thats why :sweat_smile: